src/custom/utils/blocks.ts

```diff
@@ -0,0 +1,18 @@
+import { Web3Provider } from '@ethersproject/providers'
+import { Log } from '@ethersproject/abstract-provider'
+
+export const buildBlock2DateMap = async (library: Web3Provider, logs: Log[]): Promise<Record<string, Date>> => {
+  // only check unique blocks
+  // hashes are more stable than numbers in case of reorg
+  const blockHashes = Array.from(new Set(logs.map(log => log.blockHash)))
```

Comment:
this could be expanded out a bit

Reply:
also maybe i'm missing it but why make a `Set` from the array and then make it an Array again?

Reply:
i see,

i just meant sth like
```ts
const dedupedLogs = new Set(logs.map(log => log.blockHash))
const blockHashes = Array.from(dedupedLogs)
```

Reply:
Oh, that, sure, will do